Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -57,9 +59,134 @@ namespace AzAssetBrowserRequestHandlerPrivate
+        }
+
+        AZStd::string parentName = parentSource->GetName();
+        AZStd::string parentNameWithoutExtension;
+        AzFramework::StringFunc::Path::GetFileName(parentName.c_str(), parentNameWithoutExtension);
+
+        // if we get here, we know that all the entries have the same source parent and are thus related to each other.
+        // in this case, we can run the heuristic.
+        // sort the valid entries so that they are arranged from most preferred to least preferred
+        // asset to spawn a component for:
+
+        AZStd::sort(
+            validEntries.begin(),
+            validEntries.end(),
+            [&](const ProductAssetBrowserEntry* a, const ProductAssetBrowserEntry* b)
+            {
+                // first, we use the priority system:
+                int sortPriorityA = 0;
+                int sortPriorityB = 0;
+                AZ::AssetTypeInfoBus::EventResult(
+                    sortPriorityA, a->GetAssetType(),
+                    &AZ::AssetTypeInfo::GetAssetTypeDragAndDropCreationPriority);
+                AZ::AssetTypeInfoBus::EventResult(
+                    sortPriorityB, b->GetAssetType(),
+                    &AZ::AssetTypeInfo::GetAssetTypeDragAndDropCreationPriority);
+
+                if (sortPriorityA > sortPriorityB)
+                {
+                    return true; // A definitely before B
+                }
+                else if (sortPriorityA < sortPriorityB)
+                {
+                    return false; // B definitely before A
+                }
+
+                // Getting here means their sort priority is equal.  When this is the case, we use a
+                // secondary sort scheme, which is to prefer the assets that have the exact
+                // same name as the parent source asset:
+               
+                AZStd::string nameAWithoutExtension;
+                AZStd::string nameBWithoutExtension;
+                AzFramework::StringFunc::Path::GetFileName(a->GetName().c_str(), nameAWithoutExtension);
+                AzFramework::StringFunc::Path::GetFileName(b->GetName().c_str(), nameBWithoutExtension);
+                
+                bool aMatches = AZ::StringFunc::Equal(nameAWithoutExtension, parentNameWithoutExtension, false);
+                bool bMatches = AZ::StringFunc::Equal(nameBWithoutExtension, parentNameWithoutExtension, false);
+                if ((aMatches)&&(!bMatches))
+                {
+                    return true; // A definitely before B
+                }
+                if ((!aMatches) && (bMatches))
+                {
+                    return false; // B definitely before A
+                }
+
+                // if nothing else, sort alphabetically.  A is before B if
+                // strcmp A, B < 0.  Otherwise A is not before B.  Use the extension
+                // here so as to eliminate ties.
+                return (azstricmp(a->GetName().c_str(), b->GetName().c_str()) < 0);
```

Comment:
If both Asset filenames of the ProductAssetBrowserEntry are stored in an AZ::IO::PathView, they can just be compared using operator<

```suggestion
                return AZ::IO::PathView(a->GetName(), AZ::IO::WindowsPathSeparator) < AZ::IO::PathView(b->GetName(), AZ::IO::WindowsPathSeparator);
```

Reply:
I'm actually semi concerned about this entire thing.  Looking at the Name entry its sometimes something like "Sphere (MODEL)" and not even sphere.azmodel.   I think I might have to actually add a "filename" or "pathname" entry to products so that we can make these assumptions its already making.